src/main/java/com/work/management/service/log/impl/LoggingServiceImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.work.management.service.log.impl;
+
+import com.work.management.service.log.LoggingService;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+@Service
+public class LoggingServiceImpl implements LoggingService {
+
+  private static final Logger logger = LoggerFactory.getLogger(LoggingServiceImpl.class);
+
+  public void logRequest(HttpServletRequest httpServletRequest, Object body) {
+    StringBuilder stringBuilder = new StringBuilder();
+    Map<String, String> parameters = buildParametersMap(httpServletRequest);
+
+    stringBuilder.append("REQUEST ");
+    stringBuilder.append("method=[").append(httpServletRequest.getMethod()).append("] ");
+    stringBuilder.append("path=[").append(httpServletRequest.getRequestURI()).append("] ");
+    stringBuilder.append("headers=[").append(buildHeadersMap(httpServletRequest)).append("] ");
+
+    if (!parameters.isEmpty()) {
+      stringBuilder.append("parameters=[").append(parameters).append("] ");
+    }
+
+    if (body != null) {
+      stringBuilder.append("body=[" + body + "]");
+    }
+
+    logger.info(stringBuilder.toString());
+  }
+
+  @Override
+  public void logResponse(HttpServletRequest httpServletRequest,
+      HttpServletResponse httpServletResponse, Object body) {
+    StringBuilder stringBuilder = new StringBuilder();
+
+    stringBuilder.append("RESPONSE ");
+    stringBuilder.append("method=[").append(httpServletRequest.getMethod()).append("] ");
+    stringBuilder.append("path=[").append(httpServletRequest.getRequestURI()).append("] ");
+    stringBuilder.append("responseHeaders=[").append(buildHeadersMap(httpServletResponse))
+        .append("] ");
+    stringBuilder.append("responseBody=[").append(body).append("] ");
+
+    logger.info(stringBuilder.toString());
+  }
+
+  private Map<String, String> buildParametersMap(HttpServletRequest httpServletRequest) {
```

Comment:
Instead of returing Map here, reuturn ImmutableMap, always remeber the rule: return should be as specific as possible but function parameter should be as generic as possible. You can put Map as a funciton parameter but not as function return type.